iac-extensions/terraform/src/test/resources/checks/newAbstractResourceCheckTest.tf

```diff
@@ -0,0 +1,34 @@
+// Check reportIfAbsent on AttributeSymbol
+
+resource "missing_attribute" "noncompliant" { // Noncompliant {{attribute is missing}}
+//       ^^^^^^^^^^^^^^^^^^^
+}
+
+resource "missing_attribute" "compliant" {
+  expected_attribute = true
+}
+
+// Check reportIfAbsent on BlockSymbol
+
+resource "missing_block" "noncompliant" { // Noncompliant {{block is missing}}
+  //     ^^^^^^^^^^^^^^^
+}
+
+resource "missing_block" "compliant" {
+  expected_block {}
+}
+
+// Check reportIf with isNotEquals on AttributeSymbol
+
+resource "attribute_reportIf" "noncompliant" {
+  attribute = "not_expected_value" // Noncompliant {{attribute has not expected value}}
+}
+
+resource "attribute_reportIf" "noncompliant" {
```

Comment:
Maybe the resource name should not be "noncompliant" if the TC itself is compliant?

Reply:
Good catch.